### PR TITLE
Fixing heroku read for organization app

### PR DESCRIPTION
**Pull request by kubek2k**
it will help in managing state when transitioning from org app to private and back.

### CODE REVIEW

**Comment by phinze** [comment]
Hi there - thanks for this!

Can you lay out for me some steps to take to smoke test this? Then I can verify it and pull it in if all looks good. :+1:

File: builtin/providers/heroku/resource_heroku_app.go [hunk 1]
```diff
@@ -261,12 +308,18 @@ func resourceHerokuAppRead(d *schema.ResourceData, meta interface{}) error {
...
-	d.Set("stack", app.App.Stack.Name)
-	d.Set("region", app.App.Region.Name)
+	d.Set("stack", app.App.Stack)
+	d.Set("region", app.App.Region)
 	d.Set("git_url", app.App.GitURL)
 	d.Set("web_url", app.App.WebURL)
 	d.Set("config_vars", configVarsValue)
 	d.Set("all_config_vars", app.Vars)
+	if organizationApp {
+		d.Set("organization.#", "1")
```
Comment (kubek2k) [inline builtin/providers/heroku/resource_heroku_app.go]: this line could be dropped with recent fix for the map (count no longer stored)

**Comment by kubek2k** [comment]
the situation might occur eg:
 * you have an existing private application in heroku with a name x
 * you instruct terraform to create an organization application under the same name x
terraform will conclude that the application already exists even though its in totally different namespace.

**Comment by phinze** [comment]
Working with Heroku Support to set up an organization for Terraform acceptance testing. Will follow up when I've heard back from them.

**Comment by josephholsten** [comment]
@phinze you heard back from heroku? anything we can do to move this forward?

**Comment by phinze** [comment]
I had a long and meandering support ticket that resulted in a test organization being allocated, but now logging into my account today it's gone again. :disappointed: 

Re-upped with a fresh support ticket and we'll see what happens.

If anybody from the community w/ an account can test this and report :+1: I'm happy to pull it in on your word. Otherwise I'll check back in when I get a fresh org acct.

**Comment by phinze** [comment]
Re-submitted this as #4495 - finally landing it! :sweat_smile:

**Pull request closed without merging**